### PR TITLE
common: Add pooled video frame allocator

### PR DESCRIPTION
**Pull request by LindaYu17**
Signed-off-by: Linda Yu <linda.yu@intel.com>

### CODE REVIEW

File: common/videopool.h [hunk 1]
```diff
@@ -25,11 +25,12 @@ template <class T>
 class VideoPool : public EnableSharedFromThis<VideoPool<T> >
 {
 public:
-    static SharedPtr<VideoPool<T> >
-    create(std::deque<SharedPtr<T> >& buffers)
+    VideoPool(std::deque<SharedPtr<T> >& buffers)
```
Comment (xuguangxin) [inline common/videopool.h:28]: Please split this to other commit

File: common/Makefile.am [hunk 2]
```diff
@@ -3,10 +3,12 @@ libyami_common_source_c = \
 	utils.cpp \
 	nalreader.cpp \
 	surfacepool.cpp \
+	pooledframeallocator.cpp \
```
Comment (xuguangxin) [inline common/Makefile.am]: Please use PooledFrameAllocator.cpp. We should follow webkit style https://webkit.org/code-style-guidelines/ for new file

File: common/pooledframeallocator.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef POOLED_FRAME_ALLOCATOR_H_
+#define POOLED_FRAME_ALLOCATOR_H_
```
Comment (xuguangxin) [inline common/pooledframeallocator.h]: use PooledFrameAllocator_h, it's webkit style

**Review by xuguangxin — CHANGES_REQUESTED** [review]

**Comment by LindaYu17** [comment]
change done @xuguangxin

**Review by xuguangxin — APPROVED** [review]

File: common/PooledFrameAllocator.cpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <PooledFrameAllocator.h>
```
Comment (xuguangxin) [inline common/PooledFrameAllocator.cpp]: Sorry I missed this, This should be "" instead of <>

**Comment by LindaYu17** [comment]
done @xuguangxin

File: common/PooledFrameAllocator.cpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            &attrib, 1);
+
+    if (status != VA_STATUS_SUCCESS) {
+        ERROR("create surface failed, %s", vaErrorStr(status));
+        return false;
+    }
+    std::deque<SharedPtr<VideoFrame> > buffers;
+    for (size_t i = 0;  i < surfaces.size(); i++) {
+        SharedPtr<VideoFrame> f(new VideoFrame);
+        memset(f.get(), 0, sizeof(VideoFrame));
```
Comment (chivakker) [inline common/PooledFrameAllocator.cpp:74]: This memset can be moved to the constructor of VideoFrame.  There's a similar problem with the MiscBuffer that is fixed in that way.

**Review by chivakker — APPROVED** [review]
I only have one comment but it minor at this point.  If submitter sees more benefit on pushing the memset to the constructor please do.

File: common/PooledFrameAllocator.cpp (hunk 5 above)
Comment (xuguangxin) [reply common/PooledFrameAllocator.cpp:74]: VideoFrame do not have constructor. It is a pod type; it will be use in capi too.

Comment (xuguangxin) [reply common/PooledFrameAllocator.cpp:74]: Could give more detail description for MiscBuffer thing?

**Comment by xuguangxin** [comment]
merged as 9c7bb50639233a505599a17d74caa4a0dd0ee0ad

**Pull request closed without merging**